src/main/java/com/aws/greengrass/util/Utils.java

```diff
@@ -193,15 +193,28 @@ public static String getUltimateMessage(Throwable t) {
      * @return String chain of exceptions and messages.
      */
     public static String generateFailureMessage(Throwable t) {
-        StringBuilder failureMessage = new StringBuilder(t.getMessage());
+        if (t == null) {
+            return "No error, null throwable";
+        }
+        StringBuilder failureMessage = new StringBuilder();
+        getMessageFromThrowable(t, failureMessage);
         Throwable temp = t;
         while (temp.getCause() != null) {
             temp = temp.getCause();
-            failureMessage.append(". ").append(temp.getMessage());
+            failureMessage.append(". ");
+            getMessageFromThrowable(temp, failureMessage);
         }
         return failureMessage.toString();
     }
 
+    private static void getMessageFromThrowable(Throwable t, StringBuilder builder) {
```

Comment:
nit: why not return String so you don't need to pass in the builder?

Reply:
Yeah thanks, good call. Done.